apps/webapp/app/presenters/TestJobPresenter.server.ts

```diff
@@ -139,13 +139,18 @@ export class TestJobPresenter {
         ),
       })),
       examples,
-      runs: job.runs.map((r) => ({
-        id: r.id,
-        number: r.number,
-        status: r.status,
-        created: r.createdAt,
-        payload: r.event.payload ? JSON.stringify(r.event.payload, null, 2) : undefined,
-      })),
+      runs: job.runs.map((r) => {
+        // For compatibility with old Job Runs where payload is only available on the related Event Record
+        const payload = r.payload !== null ? JSON.parse(r.payload) : r.event.payload;
```

Comment:
We'll be able to simplify this in a lot of places once we migrate existing data and make JobRun.payload non-optional.